addons/website_event_track/tests/test_website.py

```diff
@@ -19,3 +23,59 @@ def test_event_app_name(self):
         self.assertEqual(website2.events_app_name, 'Foo Events')
         website2.write({'name': 'Bar'})
         self.assertEqual(website2.events_app_name, 'Foo Events')
+
+    def test_compute_app_icon(self):
+
+        # Generate image data for JPEG
+        jpeg_image = Image.new('RGB', (60, 30), color=(73, 109, 137))
+        jpeg_io = BytesIO()
+        jpeg_image.save(jpeg_io, format='JPEG')
+        jpeg_image_data = jpeg_io.getvalue()
+
+        # Generate image data for JPG
+        jpg_image = Image.new('RGB', (60, 30), color=(73, 109, 137))
+        jpg_io = BytesIO()
+        jpg_image.save(jpg_io, format='JPEG')
+        jpg_image_data = jpg_io.getvalue()
+
+        # Generate image data for PNG
+        png_image = Image.new('RGB', (60, 30), color=(73, 109, 137))
+        png_io = BytesIO()
+        png_image.save(png_io, format='PNG')
+        png_image_data = png_io.getvalue()
+
+        # Generate image data for SVG
+        svg_image_data = b"""<svg xmlns="http://www.w3.org/2000/svg" width="60" height="30" version="1.1">
+                            <rect width="100%" height="100%" fill="rgb(73, 109, 137)"/>
+                            </svg>
+                        """
+        # Image data and their respective expected types
+        image_data = {
+            'png': png_image_data,
+            'jpg': jpg_image_data,
+            'jpeg': jpeg_image_data,
+            'svg': svg_image_data
+        }
+
+        for expected_type, image_data in image_data.items():
+            # Create a website record
+            website = self.env['website'].create({
+                'name': 'Test Website',
+                'favicon': base64.b64encode(image_data)
+            })
+
+            # Call the method to compute app_icon
+            website._compute_app_icon()
+
+            if expected_type in ['jpeg', 'png', 'jpg']:
+                # Check if app_icon is set
+                self.assertTrue(website.app_icon)
+
+                # Check if app_icon is a valid image
+                decoded_image = base64.b64decode(website.app_icon)
+                image_io = BytesIO(decoded_image)
+                image = Image.open(image_io)
```

Comment:
PIL Image has a `frombytes` method so we avoid using system calls

Reply:
Hello @thiryr,
The frombytes method in PIL has limitations as it lacks the required format attribute for comparing with the expected type.

Reply:
Oh I see, that sucks. Looks like we have a util that does that `binary_to_image`, at worst it's good to use a tool, at best we don't need a security override. Win Win.

Reply:
We have the option to achieve this task without utilizing Image.open(), but to do so, we'll need to import the imghdr module.

Reply:
Thank you for your suggestion :smile: